pkg/apiserver/rest/usecase/authentication.go

```diff
@@ -288,6 +268,107 @@ func (a *authenticationUsecaseImpl) GetDexConfig(ctx context.Context) (*apisv1.D
 	}, nil
 }
 
+func (a *authenticationUsecaseImpl) UpdateDexConfig(ctx context.Context) error {
+	connectors, err := utils.GetDexConnectors(ctx, a.kubeClient)
+	if err != nil {
+		return err
+	}
+	dexConfigApp := &v1beta1.Application{}
+	if err := a.kubeClient.Get(ctx, types.NamespacedName{
+		Name:      keyDexConfig,
+		Namespace: velatypes.DefaultKubeVelaNS,
+	}, dexConfigApp); err != nil {
+		if kerrors.IsNotFound(err) {
+			return bcode.ErrDexConfigNotFound
+		}
+		return err
+	}
+	for i, comp := range dexConfigApp.Spec.Components {
```

Comment:
Why do we not make the Dex config as a component in the Dex addon application.

Reply:
The component in addon app is not a good choice for users to edit, so make it a single application for users to modify.